music_assistant/providers/musiccast/__init__.py

```diff
@@ -0,0 +1,840 @@
+                    and device.media_title in _names
+                )
+            update_helper = UpnpUpdateHelper(
+                last_poll=now,
+                controlled_by_mass=controlled_by_mass,
+                current_queue_item_id=_queue_item_id,
+                current_uri=_player_current_url,
+            )
+
+            self.upnp_update_helper[player.player_id] = update_helper
+
+        # UPDATE PLAYBACK INFORMATION
+        if (
+            update_helper.current_queue_item_id is not None
+            and update_helper.current_uri is not None
+            and update_helper.controlled_by_mass
+        ):
+            player.set_current_media(
+                uri=update_helper.current_uri, queue_item_id=update_helper.current_queue_item_id
+            )
+        elif device.is_client:
+            _server = device.group_server
+            _server_id = self._get_player_id_from_mc_zone_player(_server)
+            _server_update_helper = self.upnp_update_helper.get(_server_id)
+            if (
+                _server_update_helper is not None
+                and _server_update_helper.current_queue_item_id is not None
+                and _server_update_helper.current_uri is not None
+                and _server_update_helper.controlled_by_mass
+            ):
+                player.set_current_media(
+                    uri=_server_update_helper.current_uri,
+                    queue_item_id=_server_update_helper.current_queue_item_id,
+                )
+            else:
+                player.set_current_media(
+                    uri=f"{_server_id}_{_server.source_id}",
+                    title=_server.media_title,
+                    artist=_server.media_artist,
+                    album=_server.media_album_name,
+                    image_url=_server.media_image_url,
+                )
+        else:
+            player.set_current_media(
+                uri=f"{player.player_id}_{device.source_id}",
+                title=device.media_title,
+                artist=device.media_artist,
+                album=device.media_album_name,
+                image_url=device.media_image_url,
+            )
+
+        # SOURCE
+        player.active_source = device.source_id  # fallback
+        if update_helper.controlled_by_mass and queue is not None:
+            # controlled by ma means directly controlled, not group member
+            player.active_source = queue.queue_id
+        elif device.is_client:
+            _server_id = self._get_player_id_from_mc_zone_player(device.group_server)
+            _server_update_helper = self.upnp_update_helper.get(_server_id)
+            _server_queue = self.mass.player_queues.get_active_queue(_server_id)
+            if (
+                _server_update_helper is not None
+                and _server_queue is not None
+                and _server_update_helper.controlled_by_mass
+            ):
+                player.active_source = _server_queue.queue_id
```

Comment:
```suggestion
```
You can leave all this out, active_source will be set for you. If you can not figure it out, just set it to None